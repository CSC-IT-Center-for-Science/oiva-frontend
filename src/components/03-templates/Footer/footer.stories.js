import React from "react";
import { storiesOf } from "@storybook/react";
import Footer from ".";
import { withInfo } from "@storybook/addon-info";

storiesOf("Footer", module)
  .addDecorator(withInfo)
  .add("Renders with the correct layout", () => <Footer />);
