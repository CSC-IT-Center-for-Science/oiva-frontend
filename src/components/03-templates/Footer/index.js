import React from "react";
import { useIntl } from "react-intl";
import logo_fi from "../../../static/images/okm-logo.svg";
import logo_sv from "../../../static/images/OKM_Sve_1rivi_logot_ISO.jpg";
import common from "../../../i18n/definitions/common";
import { Link } from "react-router-dom";
import { localizeRouteKey } from "utils/common";
import { AppRoute } from "const/index";

const Footer = () => {
  const { formatMessage, locale } = useIntl();

  const links = (
    <div className="mt-8 md:flex justify-between lg:justify-start">
      <p className="lg:mr-10">
        <Link
          to={localizeRouteKey(locale, AppRoute.Yhteydenotto, formatMessage)}
          className="underline text-green-500">
          {formatMessage(common.yhteydenotto)}
        </Link>
      </p>
      <p>
        <Link
          to={localizeRouteKey(
            locale,
            AppRoute.Saavutettavuusseloste,
            formatMessage
          )}
          className="underline text-green-500">
          {formatMessage(common.saavutettavuusseloste)}
        </Link>
      </p>
    </div>
  );

  return (
    <div className="flex justify-center lg:justify-start bg-white border-gray-300 border-t pt-12 pl-12 pr-12 pb-16">
      <div className="flex flex-col items-baseline lg:flex-1 lg:flex-row max-w-8xl mx-auto">
        <div className="flex flex-1 flex-col justify-center text-center lg:text-left lg:pr-8">
          <img
            alt={`${formatMessage(common.opetusJaKulttuuriministerio)} logo`}
            src={locale === "sv" ? logo_sv : logo_fi}
            className="max-w-sm"
          />
          {/* Visible on mobile screen size */}
          <div className="sm:hidden">
            <p>{formatMessage(common.okmAddress)} | </p>
            <p className="mt-1">
              {formatMessage(common.phoneNumber)} |{" "}
              <a
                href={formatMessage(common.okmLinkUrl)}
                className="text-green-500">
                www.minedu.fi
              </a>
            </p>
          </div>
          {/* Visible on breakpoint sm and bigger */}
          <div className="hidden sm:block">
            <p>
              {formatMessage(common.okmAddress)} |{" "}
              {formatMessage(common.phoneNumber)} |{" "}
              <a
                href={formatMessage(common.okmLinkUrl)}
                className="text-green-500 underline">
                {formatMessage(common.okmLinkText)}
              </a>
            </p>
          </div>
          <div className="block lg:hidden">{links}</div>
        </div>
        <div className="hidden lg:flex justify-end lg:block lg:flex-1 pl-8">
          {links}
        </div>
      </div>
    </div>
  );
};

export default Footer;
