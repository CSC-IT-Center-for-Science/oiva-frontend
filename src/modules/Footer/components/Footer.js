import React, { Component } from 'react'
import styled from 'styled-components'
import { COLORS, Leijona } from 'modules/styles'

import Button from '@material-ui/core/Button'

const FooterBar = styled.div`
  background: ${COLORS.WHITE};
  color: ${COLORS.BLACK};
  border-top: 0.1rem solid ${COLORS.OIVA_MEDIUM_GREEN};
`

const FooterBarUpper = styled.div`
  background: ${COLORS.WHITE};
`

class Footer extends Component {

  render() {

    return (
      <FooterBar className="flex justify-between h-auto p-2 pl-8 c-b">
        <FooterBarUpper className="flex mx-auto w-full text-xs max-w-xl">
          <div className="hidden sm:inline">
            <Leijona/>
          </div>
          <div className="pl-6 py-6">
            {/* TODO: Lokalisointi julkisille näkymille */}
            <p className="my-2">OPETUS- JA KULTTUURIMINISTERIÖ</p>
            <p className="my-2">PL 29, 00023 VALTIONEUVOSTO</p>
            <p className="my-0">PUH. 029 533 0004, <a href="http://www.minedu.fi">WWW.MINEDU.FI</a></p>
          </div>
          <Button variant="raised" color="primary">Jei</Button>
        </FooterBarUpper>
      </FooterBar>
    )
  }
}
export default Footer