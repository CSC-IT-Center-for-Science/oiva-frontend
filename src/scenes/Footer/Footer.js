import React from "react";
import { useIntl } from "react-intl";
import logo_fi from "../../static/images/okm-logo.svg";
import logo_sv from "../../static/images/OKM_Sve_1rivi_logot_ISO.jpg";
import common from "../../i18n/definitions/common";
import { Link } from "react-router-dom";

const Footer = () => {
  const intl = useIntl();

  const links = (
    <div className="mt-8 md:flex justify-between lg:justify-start">
      <p className="lg:mr-10">
        <Link to="/tietosuojailmoitus" className="underline text-green-500">
          {intl.formatMessage(common.tietosuojailmoitus)}
        </Link>
      </p>
      <p className="lg:mr-10">
        <Link to="/yhteydenotto" className="underline text-green-500">
          {intl.formatMessage(common.yhteydenotto)}
        </Link>
      </p>
      <p>
        <Link to="/saavutettavuusseloste" className="underline text-green-500">
          {intl.formatMessage(common.saavutettavuusseloste)}
        </Link>
      </p>
    </div>
  );

  return (
    <div className="flex justify-center lg:justify-start bg-white border-green-600 border-t-2 pt-12 pl-12 pr-12 pb-16">
      <div className="flex flex-col items-baseline lg:flex-1 lg:flex-row">
        <div className="flex flex-1 flex-col justify-center text-center lg:text-left lg:pr-8">
          <img
            alt="logo"
            src={intl.locale === "sv" ? logo_sv : logo_fi}
            className="lg:w-fit-content max-w-sm"
          />
          {/* Visible on mobile screen size */}
          <div className="sm:hidden">
            <p>{intl.formatMessage(common.okmAddress)} | </p>
            <p className="mt-1">
              {intl.formatMessage(common.phoneNumber)} |{" "}
              <a
                href={intl.formatMessage(common.okmLinkUrl)}
                className="text-green-500">
                www.minedu.fi
              </a>
            </p>
          </div>
          {/* Visible on breakpoint sm and bigger */}
          <div className="hidden sm:block">
            <p>
              {intl.formatMessage(common.okmAddress)} |{" "}
              {intl.formatMessage(common.phoneNumber)} |{" "}
              <a
                href={intl.formatMessage(common.okmLinkUrl)}
                className="text-green-500">
                {intl.formatMessage(common.okmLinkText)}
              </a>
            </p>
          </div>
          <div className="block lg:hidden">{links}</div>
        </div>
        <div className="hidden lg:flex justify-center lg:block lg:flex-1 pl-8">
          {links}
        </div>
      </div>
    </div>
  );
};

export default Footer;
