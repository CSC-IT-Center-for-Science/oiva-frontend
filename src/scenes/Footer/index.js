import { connect } from "react-redux";

import Footer from "./components/Footer";

const mapStateToProps = ({ user }) => ({
  user: user.user
});

export default connect(
  mapStateToProps,
  null,
  null,
  { pure: false }
)(Footer);
