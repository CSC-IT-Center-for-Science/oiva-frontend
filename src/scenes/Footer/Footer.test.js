import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Footer from './Footer';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Footer />, div);
});

it('renders a snapshot', () => {
    const tree = renderer.create(<Footer/>).toJSON();
    expect(tree).toMatchSnapshot();
  });