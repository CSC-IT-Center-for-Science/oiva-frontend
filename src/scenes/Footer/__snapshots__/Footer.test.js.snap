// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a snapshot 1`] = `
<div
  className="flex justify-center bg-white border-green-600 border-t-2 text-xxs p-4"
>
  <div
    className="flex flex-col sm:flex-row"
  >
    <div>
      <div
        className="sc-htpNat bZEFvH"
      />
    </div>
    <div
      className="flex flex-col justify-center sm:ml-10 text-center sm:text-left"
    >
      <p
        className="mt-2 pt-1 mb-1"
      >
        OPETUS- JA KULTTUURIMINISTERIOÌˆ
      </p>
      <p>
        PL 29, 00023 VALTIONEUVOSTO
      </p>
      <p
        className="mt-1"
      >
        PUH. 029 533 0004, 
        <a
          href="http://www.minedu.fi"
        >
          WWW.MINEDU.FI
        </a>
      </p>
    </div>
  </div>
</div>
`;
